### ISSUE TITLE
CrossEntropyLoss2d

### ISSUE THREAD

**Issue by kshpv** [anheidelonghu/ACNet#14]
Hello!
Thank you for your work!
I am a little bit confused about **CrossEntropyLoss2d** class, especially:

```
targets_m[mask] -= 1
loss_all = self.ce_loss(inputs, targets_m.long())
losses.append(torch.sum(torch.masked_select(loss_all, mask)) / torch.sum(mask.float()))
```
Why do you subtract 1 from targets masks?
Why do you use `torch.masked_select(loss_all, mask))` for calculating a loss for a single target? It seems incorrect because in that case we forget about a background.
Thank you!

**Comment by kshpv** [comment]
By the way, Why do you add **1** to all class labels in `                grid_image = make_grid(utils.color_label(torch.max(pred_scales[0][:3], 1)[1] + 1), 3, normalize=False,
                                       range=(0, 255))`?
Is it correct?

Thank you!

**Comment by Jinczhg** [comment]
And as for subtracting 1, I think it's just mapping class ID [1, N] to [0, N-1]. Target has to span from 0 to C. [https://pytorch.org/docs/master/generated/torch.nn.CrossEntropyLoss.html](url)
Same idea for adding back 1 to obtain the real label ID.

For the second question, Idk if this is what you wanted to ask but I'll just make a comment based on my understanding. I guess for the dataset, they don't actually have a background class, perhaps they basically assume that every object in the scene is an object that they are interested in. You can refer to the dataset document for a detailed description.
